mono/metadata/threads.c

```diff
@@ -1514,6 +1512,18 @@ thread_handle_to_internal_ptr (MonoThreadObjectHandle thread_handle)
 	return MONO_HANDLE_GETVAL(thread_handle, internal_thread); // InternalThreads are always pinned.
 }
 
+static void
+mono_error_set_exception_thread_state (MonoError *error, const char *exception_message)
```

Comment:
I usually try to make new `mono_error_set_FOO` functions varargs (calling `mono_error_set_generic_errorv`) so that the call site can have an easy way to put more context into the messages.

Reply:
None of them provide such information currently, though it could be added later.